code/modules/mob/holder.dm

```diff
@@ -25,40 +29,49 @@
 
 /obj/item/weapon/holder/process()
 
-	if(!get_holding_mob())
-
-		for(var/mob/M in contents)
+	if (!(istype(loc,/obj/item/weapon/storage)))//Mobs bug out if placed directly into a container
+		if(!get_holding_mob())
 
-			var/atom/movable/mob_container
-			mob_container = M
-			mob_container.loc = src.loc//if the holder was placed into a disposal, this should place the animal in the disposal
-			//mob_container.forceMove(get_turf(src))
-			M.reset_view()
+			for(var/mob/M in contents)
 
-		qdel(src)
+				var/atom/movable/mob_container
+				mob_container = M
+				mob_container.loc = src.loc//if the holder was placed into a disposal, this should place the animal in the disposal
+				M.reset_view()
+				M.verbs -= /mob/living/proc/get_holder_location
+			qdel(src)
```

Comment:
Shouldn't you also force a return in the process() proc, as contained may be non-existent for the next check?